File: BOJ/2078.cpp
#include <stdio.h>
#include <algorithm>
using namespace std;
int n, m;
int left, right;
int main() {
	scanf("%d%d", &n, &m);
	int a = 1;
	int b = 1;
	while (1) {
		if (n == a && m == b) break;
		if (n < m) {
			m -= n;
			right++;
		}
		else if (n>m) {
			n -= m;
			left++;
		}
		if (n == a && m != b) {
			right += (m - 1);
			m -= (m - 1);
		}
		else if (n != a && m == b) {
			left+= (n - 1);
			n -= (n - 1);
		}
	}
	printf("%d %d\n", left, right);
	return 0;
}

File: BOJ(Java 공부용)/2078.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 1, -1, 0};
    static int[] goy = {1, 0, 0, -1};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken()), m = Integer.parseInt(st.nextToken());
        int left = 0, right = 0;
        while (n != 1 || m != 1) {
            if (n < m) {
                m -= n;
                right++;
            } else if (n > m) {
                n -= m;
                left++;
            }

            if (n == 1 && m != 1) {
                right += (m - 1);
                m = 1;
            }
            if (n != 1 && m == 1) {
                left += (n - 1);
                n = 1;
            }
        }
        bw.write(left + " " + right + "");
        bw.flush();
        bw.close();
    }
}